{"primaryContentSections":[{"declarations":[{"tokens":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"cancel","kind":"identifier"},{"text":"(","kind":"text"},{"text":"promise","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"EventLoopPromise","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/EventLoopPromise","kind":"typeIdentifier","preciseIdentifier":"s:7NIOCore16EventLoopPromiseV"},{"text":"<","kind":"text"},{"text":"Void","kind":"typeIdentifier","preciseIdentifier":"s:s4Voida"},{"text":">? = nil)","kind":"text"}],"platforms":["Linux"],"languages":["swift"]}],"kind":"declarations"},{"content":[{"type":"heading","text":"Discussion","level":2,"anchor":"discussion"},{"inlineContent":[{"text":"Whether the execution of the task is immediately canceled depends on whether the execution of a task has already begun.","type":"text"},{"text":" ","type":"text"},{"text":"This means immediate cancellation is not guaranteed.","type":"text"}],"type":"paragraph"},{"inlineContent":[{"text":"The safest way to cancel is by using the passed reference of ","type":"text"},{"type":"codeVoice","code":"RepeatedTask"},{"text":" inside the task closure.","type":"text"}],"type":"paragraph"},{"inlineContent":[{"text":"If the promise parameter is not ","type":"text"},{"code":"nil","type":"codeVoice"},{"text":", the passed promise is fulfilled when cancellation is complete.","type":"text"},{"text":" ","type":"text"},{"text":"Passing a promise does not prevent fulfillment of any promise provided on original task creation.","type":"text"}],"type":"paragraph"}],"kind":"content"}],"kind":"symbol","sections":[],"schemaVersion":{"patch":0,"major":0,"minor":3},"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/niocore\/repeatedtask\/cancel(promise:)"]}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/RepeatedTask\/cancel(promise:)"},"metadata":{"fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"externalParam","text":"promise"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:7NIOCore16EventLoopPromiseV","text":"EventLoopPromise"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","preciseIdentifier":"s:s4Voida","text":"Void"},{"kind":"text","text":">?)"}],"modules":[{"name":"NIOCore"}],"roleHeading":"Instance Method","title":"cancel(promise:)","externalID":"s:7NIOCore12RepeatedTaskC6cancel7promiseyAA16EventLoopPromiseVyytGSg_tF","symbolKind":"method","role":"symbol"},"abstract":[{"text":"Try to cancel the execution of the repeated task.","type":"text"}],"hierarchy":{"paths":[["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/RepeatedTask"]]},"references":{"doc://co.pointfree.VaporRouting/documentation/NIOCore/RepeatedTask":{"role":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/RepeatedTask","kind":"symbol","fragments":[{"kind":"keyword","text":"class"},{"kind":"text","text":" "},{"kind":"identifier","text":"RepeatedTask"}],"navigatorTitle":[{"kind":"identifier","text":"RepeatedTask"}],"abstract":[{"type":"text","text":"Returned once a task was scheduled to be repeatedly executed on the "},{"code":"EventLoop","type":"codeVoice"},{"type":"text","text":"."}],"type":"topic","url":"\/documentation\/niocore\/repeatedtask","title":"RepeatedTask"},"doc://co.pointfree.VaporRouting/documentation/NIOCore":{"url":"\/documentation\/niocore","title":"NIOCore","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore","kind":"symbol","type":"topic","abstract":[],"role":"collection"},"doc://co.pointfree.VaporRouting/documentation/NIOCore/EventLoopPromise":{"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/EventLoopPromise","role":"symbol","kind":"symbol","url":"\/documentation\/niocore\/eventlooppromise","abstract":[{"type":"text","text":"A promise to provide a result later."}],"title":"EventLoopPromise","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"EventLoopPromise"}],"type":"topic","navigatorTitle":[{"kind":"identifier","text":"EventLoopPromise"}]},"doc://co.pointfree.VaporRouting/documentation/NIOCore/RepeatedTask/cancel(promise:)":{"role":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/RepeatedTask\/cancel(promise:)","kind":"symbol","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"cancel","kind":"identifier"},{"text":"(","kind":"text"},{"text":"promise","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"EventLoopPromise","kind":"typeIdentifier","preciseIdentifier":"s:7NIOCore16EventLoopPromiseV"},{"text":"<","kind":"text"},{"text":"Void","kind":"typeIdentifier","preciseIdentifier":"s:s4Voida"},{"text":">?)","kind":"text"}],"abstract":[{"text":"Try to cancel the execution of the repeated task.","type":"text"}],"type":"topic","url":"\/documentation\/niocore\/repeatedtask\/cancel(promise:)","title":"cancel(promise:)"}}}